### PR TITLE
Handle new 'invalid SCC URL' pop-up

### PR DESCRIPTION
**Pull request by dzedro**
https://trello.com/c/N09TRZxX/968-3-don-t-crash-on-invalid-regurl-on-linuxrc-commandline

[textmode invalid URL test](http://10.100.12.155/tests/183#step/welcome/1)
[gnome invalid URL test](http://10.100.12.155/tests/6184#step/welcome/1)

[textmode invalid URL test when SCC_URL_VALID is not defind, in some random test](http://10.100.12.155/tests/6186#step/welcome/3)

[autoyast test](http://10.100.12.155/tests/6166#step/installation/2)
[casp autoyast test - detect warning and end](http://10.100.12.155/tests/6173#step/installation/1)

[gnome test](http://10.100.12.155/tests/6185#step/welcome/1)

### CODE REVIEW

File: tests/installation/welcome.pm [hunk 1]
```diff
@@ -18,36 +18,44 @@ use testapi;
...
-    else {
-        assert_screen @welcome_tags, $bootup_timeout;
-    }
-    if (match_has_tag('inst-welcome-confirm-self-update-server')) {
-        wait_screen_change { send_key $cmd{ok} };
-        assert_screen 'inst-welcome';
-    }
+    while (1) {
+        my @welcome_tags = ('inst-welcome', 'inst-welcome-confirm-self-update-server', 'scc-invalid-url');
+        push @welcome_tags, 'inst-betawarning' if get_var('BETA');
```
Comment (kravciak) [inline tests/installation/welcome.pm]: This removes check for beta license (makes it optional)

File: tests/installation/welcome.pm [hunk 2]
```diff
@@ -18,36 +18,44 @@ use testapi;
...
-        assert_screen @welcome_tags, 10;
-    }
-    else {
-        assert_screen @welcome_tags, $bootup_timeout;
-    }
-    if (match_has_tag('inst-welcome-confirm-self-update-server')) {
-        wait_screen_change { send_key $cmd{ok} };
-        assert_screen 'inst-welcome';
-    }
+    while (1) {
```
Comment (kravciak) [inline tests/installation/welcome.pm]: Why do you use cycle here? Do you expect some needle to match more than once?
Why not make it simpler by:
 - assert_screen scc-invalid-url if SCC_URL_INVALID
 - assert_screen inst-betawarning if BETA
 - assert_screen inst-welcome-confirm-self-update if UPDATE
 - assert_screen inst_welcome

Comment (dzedro) [inline tests/installation/welcome.pm]: It's not so simple as you think
any of examples you mentioned can occur without specifying it in test
I see two options use loop or create not small condition tree

File: tests/installation/welcome.pm [hunk 3]
```diff
@@ -18,36 +18,49 @@ use testapi;
...
-    ensure_fullscreen;
-    my $bootup_timeout = 500;
-    if (get_var("BETA")) {
-        assert_screen "inst-betawarning", $bootup_timeout;
-        send_key "ret";
-        assert_screen @welcome_tags, 10;
+    while (1) {
+        my @welcome_tags = ('inst-welcome', 'inst-welcome-confirm-self-update-server', 'scc-invalid-url');
+        # replace tag 'inst-welcome' with 'inst-betawarning' to avoid race condition
+        shift @welcome_tags && push @welcome_tags, 'inst-betawarning' if get_var('BETA');
```
Comment (kravciak) [inline tests/installation/welcome.pm]: I would prefer this, but functionality is the same, so ignore if you don't agree :)
if (get_var BETA)
  push @welcome_tags, 'inst-betawarning'
else
  push @welcome_tags, 'inst-welcome'

Comment (coolo) [inline tests/installation/welcome.pm]: I would prefer that too - the proposed form requires quite some knowledge about the preference of perl operators

Comment (dzedro) [inline tests/installation/welcome.pm]: yep, used the clearer version

File: tests/installation/welcome.pm [hunk 4]
```diff
@@ -18,36 +18,53 @@ use testapi;
...
+        }
+        ensure_fullscreen;
+        assert_screen \@welcome_tags, 500;
+        if (match_has_tag 'scc-invalid-url') {
+            die 'SCC reg URL is invalid' if !get_var('SCC_URL_VALID');
+            send_key 'alt-r';    # registration URL field
+            send_key_until_needlematch 'scc-invalid-url-deleted', 'backspace';
+            type_string get_var('SCC_URL_VALID');
+            wait_still_screen 2;
+            send_key 'alt-o';    # OK
```
Comment (kravciak) [inline tests/installation/welcome.pm]: why not use wait_screen_change as on line 41? (without wait_still_screen) ..just for consistency :)

**Comment by kravciak** [comment]
LGTM

File: tests/installation/welcome.pm (hunk 4 above)
Comment (dzedro) [inline tests/installation/welcome.pm]: wait_screen_change is crap, mostly is too fast, test was failing when I used it, so I used wait_still_screen
but I will use wait_screen_change with alt-o, that should be fine

**Comment by chuchingkai** [comment]
A failed case:https://openqa.suse.de/tests/955819#step/welcome/3

**Pull request merged**